src/db.rs

```diff
@@ -31,20 +32,20 @@ use std::ffi::{CStr, CString};
 use std::fmt;
 use std::fs;
 use std::iter;
-use std::marker::PhantomData;
 use std::path::Path;
 use std::path::PathBuf;
 use std::ptr;
 use std::slice;
 use std::str;
+use std::sync::Arc;
 use std::sync::RwLock;
 use std::time::Duration;
 
 // Marker trait to specify single or multi threaded column family alternations for DB
 // Also, this is minimum common API sharable between SingleThreaded and
 // MultiThreaded. Others differ in self mutability and return type.
 pub trait ThreadMode {
-    fn new(cf_map: BTreeMap<String, ColumnFamily>) -> Self;
+    fn new(cf_map: BTreeMap<String, *mut ffi::rocksdb_column_family_handle_t>) -> Self;
```

Comment:
Is it really good idea to use `ffi` types in public api ?

Reply:
certainly, it's not good idea... but we have to do that... kind of. I added some reasoning here: https://github.com/rust-rocksdb/rust-rocksdb/pull/509/commits/0d5ff023b42536aee8869fac5a504b553ff0fe4d

I googled a bit but found that it's quite bothersome to hide actual trait methods from public traits...

In short, these methods aren't intended to be consumed, nevertheless both pre- and post- this pr.

Reply:
Should the trait `ThreadMod` be public ??

Reply:
I got it. I've seen a description a bit later.